CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/CVKCanvas.java

```diff
@@ -5,83 +5,115 @@
  */
 package au.gov.asd.tac.constellation.visual.vulkan;
 
+import au.gov.asd.tac.constellation.visual.vulkan.renderables.CVKRenderable;
+import au.gov.asd.tac.constellation.visual.vulkan.utils.CVKGraphLogger;
+import static au.gov.asd.tac.constellation.visual.vulkan.utils.CVKUtils.CVKAssertNotNull;
 import java.awt.Graphics;
+
 import org.lwjgl.vulkan.awt.AWTVKCanvas;
 import javax.swing.SwingUtilities;
+import static org.lwjgl.vulkan.VK10.VK_NULL_HANDLE;
 import org.lwjgl.vulkan.awt.VKData;
 
 
-
-public class CVKCanvas extends AWTVKCanvas{
-    private boolean parentAdded = false;
-    private int frameNumber = 0;
+public class CVKCanvas extends AWTVKCanvas {
+    private final CVKVisualProcessor cvkVisualProcessor;    
     private final CVKRenderer cvkRenderer;
+    private int frameNumber = 0;
     
     public int GetFrameNumber() { return frameNumber; }
+    public CVKRenderer GetRenderer() { return cvkRenderer; }
+    public CVKGraphLogger GetLogger() { return cvkVisualProcessor.GetLogger(); }
+    public void VerifyInRenderThread() { cvkVisualProcessor.VerifyInRenderThread(); }
+    public boolean IsRenderThreadCurrent() { return cvkVisualProcessor.IsRenderThreadCurrent(); }
+    public boolean IsRenderThreadAlive() { return cvkVisualProcessor.IsRenderThreadAlive(); }
+    public void AddRenderable(CVKRenderable renderable) { cvkRenderer.AddRenderable(renderable); }
     
-    public CVKCanvas(VKData vkData, CVKRenderer cvkRenderer) {
-        super(vkData);
-        cvkRenderer.Logger().fine("Canvas constructed");        
-        this.cvkRenderer = cvkRenderer;
-        this.addComponentListener(cvkRenderer);
+    
+    @Override
+    public void paint(Graphics g) {
```

Comment:
Is there a plan to have code in this function? Does it need to be overriden if it just calls super